src/commands/external.rs

```diff
@@ -158,28 +158,46 @@ async fn consider_install(
             }
             plugin_installer.run().await?;
         }
-        Ok(None) // No update badgering needed if we just updated/installed it!
-    } else if stderr().is_terminal() && matches_catalogue_plugin(plugin_store, plugin_name) {
-        if offer_install(plugin_name)? {
-            let plugin_installer = installer_for(plugin_name);
-            plugin_installer.run().await?;
-            eprintln!();
-            Ok(None) // No update badgering needed if we just updated/installed it!
-        } else {
-            process::exit(2);
+        return Ok(None); // No update badgering needed if we just updated/installed it!
+    }
+
+    if stderr().is_terminal() {
+        if let Some(plugin) = match_catalogue_plugin(plugin_store, plugin_name) {
+            let package = spin_plugins::manager::get_package(&plugin)?;
+            if offer_install(&plugin, package)? {
+                let plugin_installer = installer_for(plugin_name);
+                plugin_installer.run().await?;
+                eprintln!();
+                return Ok(None); // No update badgering needed if we just updated/installed it!
+            } else {
+                process::exit(2);
+            }
         }
-    } else {
-        tracing::debug!("Tried to resolve {plugin_name} to plugin, got {e}");
-        terminal::error!("'{plugin_name}' is not a known Spin command. See spin --help.\n");
-        print_similar_commands(app, plugin_name);
-        process::exit(2);
     }
+
+    tracing::debug!("Tried to resolve {plugin_name} to plugin, got {e}");
+    terminal::error!("'{plugin_name}' is not a known Spin command. See spin --help.\n");
+    print_similar_commands(app, plugin_name);
+    process::exit(2);
 }
 
-fn offer_install(name: &str) -> anyhow::Result<bool, anyhow::Error> {
-    terminal::warn!("`{name}` is not a known Spin command, but there is a plugin with that name.");
+fn offer_install(
+    plugin: &spin_plugins::manifest::PluginManifest,
+    package: &spin_plugins::manifest::PluginPackage,
+) -> anyhow::Result<bool, anyhow::Error> {
+    terminal::warn!(
+        "`{}` is not a known Spin command, but there is a plugin with that name.",
+        plugin.name()
+    );
+    let prompt = format!(
+        "Would you like to install the '{}' with {} license from {}, and run it now?",
```

Comment:
When using long lines on prompts, if there is a line-wrap, the confirmation will cause it to print twice. This was an issue we had with plugins earlier that we solved by splitting it into two lines with the line on the prompt being shorter.